tests/e2e/testutils.go

```diff
@@ -420,6 +420,8 @@ func RestartCluster(nodeNames []string) error {
 	for _, nodeName := range nodeNames {
 		cmd := "systemctl restart k3s* --all"
 		if _, err := RunCmdOnNode(cmd, nodeName); err != nil {
+			journalCmd := "journalctl -u k3s"
+			RunCmdOnNode(journalCmd, nodeName)
```

Comment:
There's a `GetJournalLogs(nodeName)` function, it should probably be wired up as an on-failure action for all these tests.